packages/runtime/src/next/utils/pagination.ts

```diff
@@ -0,0 +1,54 @@
+import { AsyncIterableExt } from './async-iterable'
+
+type PaginationResult<T> = { data: T[]; hasMore: boolean }
+type IterablePaginationResult<T> = Promise<PaginationResult<T>> & AsyncIterableExt<T>
+
+type PaginationFunction<Options, R extends { id: any }> = Options extends {
+  after?: string
+}
+  ? (options: Options) => Promise<PaginationResult<R>>
+  : never
+
+type IterableMethod<Options, R> = Partial<Options> extends Options
+  ? (options?: Options) => IterablePaginationResult<R>
+  : (options: Options) => IterablePaginationResult<R>
+
+export function toIterablePaginationResult<Options, R extends { id: any }>(
```

Comment:
This function is a bit tricky to follow but I'm assuming all the type inference works as expected? Also, is there a reason not to make `id` be of type `string` instead of `any`?

Reply:
Realistically, no. We wanted this helper to be flexible, so perhaps an entity could have a numerical `id`, etc.

Reply:
In that scenario we could do `string | number` instead of `any`. Not a big deal, but `any` is a type that spreads around so I generally like to avoid it whenever possible

Reply:
Got it - then to avoid spreading I'll restrict `id` to a `string` for now.